### PR TITLE
Utilities: Add type safety to createArray and findIndex

### PR DESCRIPTION
**Pull request by christiango**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Include a change request file using `$ npm run change`

#### Description of changes
* Added unit tests to array methods
* Added typesafety to createArray and findIndex

### CODE REVIEW

File: packages/utilities/src/array.ts [hunk 1]
```diff
@@ -26,11 +26,11 @@ export function findIndex(array: any[], cb: (item: any, index?: number) => boole
  * @param size - Size of array.
  * @param getItem - Callback to populate given cell index.
  */
-export function createArray(size: number, getItem?: (index?: number) => any): any[] {
```
Comment (christiango) [inline packages/utilities/src/array.ts]: Intentional breaking change to make type safety work by making getItem required. Thoughts @dzearing ?

File: packages/office-ui-fabric-react/src/components/FocusZone/examples/FocusZone.List.Example.tsx [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { DetailsRow, IColumn, Selection, SelectionMode } from 'office-ui-fabric-
 import './FocusZone.List.Example.scss';
 
 const ITEMS = createArray(10, (index) => ({
-  key: index,
+  key: index.toString(),
```
Comment (christiango) [inline packages/office-ui-fabric-react/src/components/FocusZone/examples/FocusZone.List.Example.tsx]: Legitimate bug found by this change!

**Comment by dzearing** [comment]
Too late, i already fixed these yesterday! weird timing ;)

**Comment by dzearing** [comment]
Merging your test though

**Comment by christiango** [comment]
That is quite the coincidence. Couple of things with your version:

Why is index optional in the callback function for findIndex? The callback always provides it.

What's the scenario for createArray not using a creator function? I'd love to get rid of the null insertion and the typecast

**Pull request merged**